pkg/api/client_options.go

```diff
@@ -89,7 +89,7 @@ func optionsNeedResolution(opts ClientOptions) bool {
 }
 
 func resolveOptions(opts ClientOptions) (ClientOptions, error) {
-	cfg, _ := config.Read()
+	cfg, _ := config.Read(nil)
```

Comment:
It's kind of a shame with this pattern that we end up passing `nil` around everywhere. Oh well.

Reply:
Agreed. Open to better suggestions if you have them.

Reply:
The only reasonable one I have is using variadic functional opts. Everything else is a much bigger change than I think we would want to tackle.